### PR TITLE
Properly handling WalletConnect deep/universal links

### PR DESCRIPTION
**Pull request by AlexBHarley**
### Description

Take a look [here](https://docs.walletconnect.org/v/2.0/mobile-linking) for an overview of the problem. Basically we're conforming to the spec for deep/universal links so it's easier for other frontend libraries to integrate with Valora.
 
### Other changes

N/A

### Tested

Added unit tests.

### How others should test

Can use Plock.fi and use-contractkit.vercel.app for testing. These changes should make it smoother to use on mobile.

### Related issues

N/A

### CODE REVIEW

File: packages/mobile/src/walletConnect/walletConnect.ts [hunk 1]
```diff
@@ -1,7 +1,39 @@
 import { call } from 'redux-saga/effects'
 import { initialiseWalletConnect } from 'src/walletConnect/saga'
 
+const wcPrefix = 'wc:'
+const deepLinkPrefix = 'celo://wallet/wc?uri='
+const universalLinkPrefix = 'https://valoraapp.com/wc?uri='
```
Comment (jeanregisser) [inline packages/mobile/src/walletConnect/walletConnect.ts:6]: Currently registered universal link on iOS is `vlra.app`
https://github.com/celo-org/wallet/blob/24af3d30f89a6520190a9c78ccc4e62af57b9953/packages/mobile/ios/celo/celo.entitlements#L10
Which is used for firebase dynamics links.

We should add `valoraapp.com` there and also update the associated config on the website: https://valoraapp.co/apple-app-site-association

Comment (jeanregisser) [inline packages/mobile/src/walletConnect/walletConnect.ts:6]: nit:
```suggestion
const WC_PREFIX = 'wc:'
const DEEPLINK_PREFIX = 'celo://wallet/wc?uri='
const UNIVERSAL_LINK_PREFIX = 'https://valoraapp.com/wc?uri='
```

**Review by jeanregisser — COMMENTED** [review]
Nice! 👍

File: packages/mobile/src/walletConnect/walletConnect.ts (hunk 1 above)
Comment (AlexBHarley) [reply packages/mobile/src/walletConnect/walletConnect.ts:6]: Pinged the web team

File: packages/mobile/src/app/saga.ts [hunk 2]
```diff
@@ -193,8 +199,9 @@ export function* watchDeepLinks() {
 
 export function* handleOpenUrl(action: OpenUrlAction) {
   const { url, openExternal, isSecureOrigin } = action
+  const walletConnectEnabled: boolean = yield select(walletConnectEnabledSelector)
   Logger.debug(TAG, 'Handling url', url)
-  if (url.startsWith('celo:')) {
+  if (url.startsWith('celo:') || (url.startsWith('wc:') && walletConnectEnabled)) {
```
Comment (jeanregisser) [inline packages/mobile/src/app/saga.ts:204]:
```suggestion
  if (url.startsWith('celo:') || (walletConnectEnabled && isWalletConnectDeepLink(url))) {
```

File: packages/mobile/android/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -23,6 +23,7 @@
         <category android:name="android.intent.category.DEFAULT" />
         <category android:name="android.intent.category.BROWSABLE" />
         <data android:scheme="celo" android:host="wallet" />
+        <data android:scheme="wc" android:host="wallet" />
```
Comment (jeanregisser) [inline packages/mobile/android/app/src/main/AndroidManifest.xml:26]: Are links actually starting with `wc://wallet`?
if not we should remove the `android:host` part.

Comment (AlexBHarley) [reply packages/mobile/android/app/src/main/AndroidManifest.xml:26]: Good point, they're like `wc:123` so no need for that.

**Review by jeanregisser — APPROVED** [review]

**Comment by Celoqa** [comment]
Hey @MuckT I have verified this issue using Android internal build V1.16.1 & Test Flight build V1.16.2(68) & observed following observations:
**Observations/Query:**

- We have observed that, in Android user redirect to blank screen instead of respective wallet connect page when user taps on deep link prepared from QR code link 
- In iOS, nothing happens when user taps on deep link
**FYI:** Currently we have checked this scenario by preparing  deep link using [Code Pen Link](https://drive.google.com/file/d/1ycLjDTaRi97p23lWShORMDkzv_SSBQWY/view?usp=sharing) 
Can you please let us know needful for above observations. And also let us please know if there any other way to prepare deep link ?


**Attachments:** [DeeplinkWCIssue](https://drive.google.com/file/d/1H_g6NLjAdbrEiIbwEb0WgDLTwHZk_BVR/view?usp=sharing)

 



 
**Verified On Devices:** Redmi Note 8 (9.0),iPhone SE (14.2)

CC: @AlexBHarley 
**Thanks..!!**

**Pull request merged**